File: src/Services/AdvertisementService/Domain/Entities/Advertisement.cs
﻿using Domain.Entities;

namespace AdvertisementService.Domain.Entities;

public class Advertisement
{
    public Guid Id { get; set; }  
    public int SellingPrice { get; set; } = 0;
    public string? Seller { get; set; }  
    public string? Buyer { get; set; }
    public int? SoldAmount { get; set; }
    public int NumberOfOffers { get; set; } = 0;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdateAt { get; set; } = DateTime.UtcNow;
    public DateTime? EndedAt { get; set; }
    public AdvertisementStatus Status { get; set; } = AdvertisementStatus.Live;
    public required OfferTypePretended OfferTypePretended { get; set; }
    public required Item Item { get; set; }
}

File: frontend/web-app/types/Advertisement.ts
import { Item } from "./Item"

export type Advertisement = {
  id: string
  sellingPrice: number
  seller?: string
  buyer?: string
  soldAmount?: number
  numberOfOffers: number
  createdAt: string
  updateAt: string
  endedAt: string
  status: string
  offerTypePretended: string
  item: Item
}